### DISCUSSION TITLE
Jetbrains Rider with Copilot plugin: Failed to initiate the GitHub login process. Please try again.

### DISCUSSION THREAD

**Discussion by JohanStolk** [community/community#37993] [Copilot Conversations]
I've succesfully used Copilot plugin in Rider on other systems (windows/mac) but it is not working in my new work environment. (windows 10)

Keep getting: 

Failed to initiate the GitHub login process. Please try again.

Tried re-installing, restarting, etc. 
Still not working, I've paid for copilot but cannot use it in my new work environment.

How to fix this?

**Comment by iamcjwilliams** [comment]
@JohanStolk Hello, can you provide a screenshot of the error you're receiving by performing  (View -> Output; select GitHub Copilot from the pulldown)

**Reply by JohanStolk** [reply]
I've attached screenshots in the post below, thanks.

**Comment by JohanStolk** [comment]
Hi, thanks for the response.
In my Rider version there is no View => Output
I've attached screenshots showing my layout and results
<img width="2880" alt="Screenshot 2022-11-03 085422" src="https://user-images.githubusercontent.com/3437419/199671298-d06855c7-e8af-48c0-9381-e686eaa474df.png">
<img width="2880" alt="Screenshot 2022-11-03 085741" src="https://user-images.githubusercontent.com/3437419/199671315-b127416d-3b32-4479-8341-397d89e2d871.png">

**Reply by iamcjwilliams** [reply]
Are you using the latest version? Also, have you tried to logout of your rider account and your GitHub account in a web browser then logging in again?

**Reply by JohanStolk** [reply]
Yes, latest versions of Rider and plugin. Logged out and in of JB account, no effect. Github login in browser works fine but I get the error (dialog) directly, no chance of logging in. Are there any log files to see where it goes wrong? My company uses Zscaler for security, could that be related?

**Reply by JohanStolk** [reply]
I've found this https://github.com/orgs/community/discussions/29127#discussioncomment-3981077
but no idea how to get proxy settings here
given the security here http is unlikely
Copilot is great but why is using it behind VPN not easily supported? Unexpected and already paid for 1 year...

**Comment by Moses-Code-Dev** [comment]
https://github.com/djoxy-code/Github-copilot-Fix

**Reply by JohanStolk** [reply]
Thanks for the suggestion, for security reasons I'm reluctant to add the jars...
I've updated to latest version of node (18.12.1 LTS) but no effect...

**Comment by JohanStolk** [comment]
Still stuck with this issue so I tried it in Visual Studio, the error I got is this: unable to get local issuer certificate
See full error message below.
Probably related with Rider plugin issue but that does not give me any logs...

GitHub Copilot 1.56.0.2 (v1.56.0.2@97bf1be2b)
GitHub Copilot Agent v1.56.0
[Error] An error occurred for 'RunOperation:SignInInitiateAsync': StreamJsonRpc.RemoteInvocationException: Request signInInitiate failed with message: unable to get local issuer certificate
   at StreamJsonRpc.JsonRpc.<InvokeCoreAsync>d__143`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.Copilot.CopilotClient.<RunOperation>d__17`1.MoveNext()
[Error] An error occurred for 'RunOperation:RunOperation': StreamJsonRpc.RemoteInvocationException: Request signInInitiate failed with message: unable to get local issuer certificate
   at StreamJsonRpc.JsonRpc.<InvokeCoreAsync>d__143`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.Copilot.CopilotClient.<RunOperation>d__17`1.MoveNext()
NotSignedIn

**Reply by Moses-Code-Dev** [reply]
Try to downgrade with the method described in the earlier link,
Just make sure to replace windows path with the corresponding one on Mac if you are using Mac

**Reply by iamcjwilliams** [reply]
@JohanStolk The error message unable to get local issuer certificate means a proxy that uses self-signed certificates is being used. Unfortunately, self-signed certificates are something we currently don't support. I'd recommend that you keep an eye on our [official GitHub public product roadmap](https://github.com/github/roadmap), our [Changelog](https://github.blog/changelog/) and our [GitHub Blog](http://github.blog/) for future updates.

**Reply by JohanStolk** [reply]
I miss my copilot! Not sure I understand why licensing works fine with Jetbrains Rider and Microsoft Visual Studio but not for copilot. I hope the team can fix this ASAP. Please

**Reply by iamcjwilliams** [reply]
Our copilot team is reviewing this issue

**Reply by JohanStolk** [reply]
Thank you

**Reply by vishanu1** [reply]
Any updates on this ? 
I am also struggling to setup the copilot latest 1.2.21.**** version with IntelliJ 2023.2 community edition 
Let me know if there is any support team for the same to whom I can directly approach ? 
@iamcjwilliams - appreciate your support here

**Comment by baoge2778** [comment]
1.https://plugins.jetbrains.com/plugin/17718-github-copilot/versions/stable 下载 GitHub Copilot 1.1.28.1744，把最新版本C:\Users\{name}\AppData\Roaming\JetBrains\IntelliJIdea2022.1\plugins\github-copilot-intellij替换
2.在idea下help的vm options添加-Dcopilot.agent.disabled=true
解决问题。

**Reply by AntarcticPresident** [reply]
老哥，稳！
上面那个老外看不懂中文，就跟那抓瞎。让他不学中文

**Reply by zhasngf** [reply]
你是懂中文吧！

**Comment by RasulOs** [comment]
I have the same problem. I solved it in VSCode by using this solution: https://stackoverflow.com/questions/71367058/self-signed-certificate-in-certificate-chain-on-github-copilot .

Check, if your problem also will be solved, it means it is the same problem.

But I don't know how to solve it in Android Studio and Intellij IDEA. But I think the problem in VSCode and in Intellij or Android Studio is the same.

P.S. copilot works on my private computer but does not work on a working laptop. I disabled an antivirus for a short period of time, to see if the copilot does not work because of the antivirus, and after disabling it copilot works.

**Comment by vladamir** [comment]
Helped me on my iMac:

```
sudo rm -rf ~/.cache
mkdir ~/.cache
chmod 700 ~/.cache
```
